### PR TITLE
SI-152: Add an option to run Rackattack as docker containers

### PR DESCRIPTION
**Pull request by eliran-stratoscale**

### CODE REVIEW

**Comment by eliran-stratoscale** [comment]
@Stratoscale/systegration

File: docker/rackattack-physical-base.dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    ipmitool \
+    python-devel \
+    python-pip \
+    make \
+    sudo \
+    boost-devel \
+    boost-static \
+    openssl-devel \
+    gcc-c++ \
+    which
```
Comment (rom-stratoscale) [inline docker/rackattack-physical-base.dockerfile]: yum clean all

Comment (rom-stratoscale) [inline docker/rackattack-physical-base.dockerfile]: what they usually do is a build container and deploy container.
This way, u dont have git, python-devel, make,boost, gcc installed in the deploy docker. it should be thin.
http://blog.xebia.com/create-the-smallest-possible-docker-container/ - interesting blog

you can see how docker themselves use it:
https://github.com/docker/compose/tree/master/script/release

for now what u did is fine, but read the above links for general understanding

File: docker/rackattack-physical.dockerfile.m4 [hunk 2]
```diff
@@ -0,0 +1,25 @@
+include(rackattack-physical-base.dockerfile)
+
+# Install other tools
+RUN yum update -y && \
```
Comment (rom-stratoscale) [inline docker/rackattack-physical.dockerfile.m4]: why do u have the yum update here? you have it in your base. this layer should be thin

File: docker/rackattack-physical.dockerfile.m4 [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    yum install -y \
+    dnsmasq \
```
Comment (rom-stratoscale) [inline docker/rackattack-physical.dockerfile.m4]: why are those not part of the base. make this layer rackattack related only

File: docker/rackattack-physical-base.dockerfile (hunk 1 above)
Comment (eliran-stratoscale) [inline docker/rackattack-physical-base.dockerfile]: done (yum clean all)

File: docker/rackattack-physical.dockerfile.m4 (hunk 2 above)
Comment (eliran-stratoscale) [inline docker/rackattack-physical.dockerfile.m4]: fixed

File: docker/rackattack-physical.dockerfile.m4 (hunk 3 above)
Comment (eliran-stratoscale) [inline docker/rackattack-physical.dockerfile.m4]: fixed

**Pull request merged**